### PR TITLE
Fix dependabot PRs

### PR DESCRIPTION
**Pull request by reakaleek**
## Motivation/summary

Dependabot Pull Requests are not updating the NOTICE.txt file, which causes the lint job to fail.
This workflow will checkout the dependabot PR, update the NOTICE.txt file, and push the changes back to the PR.

## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

### CODE REVIEW

File: .github/workflows/update-dependabot-pr.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      - dependabot/go_modules/**
+
+permissions:
+  contents: read
+
+jobs:
+  update-dependbot-pr:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: elastic/apm-pipeline-library/.github/actions/github-token@current
```
Comment (reakaleek) [inline .github/workflows/update-dependabot-pr.yml:19]: @v1v 

I was also thinking.. maybe we could use the github actions token instead with

```
permissions:
  contents: write
```

But then, I guess, the tests won't run.

WDYT?

Comment (reakaleek) [reply .github/workflows/update-dependabot-pr.yml:19]: Oh wait. Maybe we actually want to use the github actions token.

pushing to the branch will cause the workflow to be triggered again and again.

Comment (v1v) [reply .github/workflows/update-dependabot-pr.yml:19]: In addition, using the `GitHub bot` will prevent the build from being rerun.

However, I wonder whether a changeset validation should be in place, so if changes other than `NOTICE`, then run the next steps.

What do you think?

Comment (reakaleek) [reply .github/workflows/update-dependabot-pr.yml:19]: Aha, 

we could do 

```
    paths-ignore:
      - NOTICE.txt
```

Then this check will only run if the `NOTICE.txt` file was not changed. Hence, this workflow won't run twice.

File: .github/workflows/update-dependabot-pr.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+      - uses: elastic/apm-pipeline-library/.github/actions/setup-git@current
+        with:
+          username: ${{ env.GIT_USER }}
+          email: ${{ env.GIT_EMAIL }}
+          token: ${{ env.GITHUB_TOKEN }}
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.branch }}
```
Comment (v1v) [inline .github/workflows/update-dependabot-pr.yml:33]: why?

Comment (reakaleek) [reply .github/workflows/update-dependabot-pr.yml]: woops.

https://github.com/elastic/apm-server/pull/12543/commits/6d12399494822c931d6773388522bf39d0a4b8d8

**Review by v1v — APPROVED** [review]

**Pull request merged**